sdk/typescript/runtime/main.go

```diff
@@ -275,8 +275,10 @@ func (t *TypescriptSdk) Base() (*dagger.Container, error) {
 			WithMountedCache("/root/.npm", dag.CacheVolume(fmt.Sprintf("npm-cache-%s-%s", runtime, version))).
 			WithMountedCache("/root/.cache/yarn", dag.CacheVolume(fmt.Sprintf("yarn-cache-%s-%s", runtime, version))).
 			WithMountedCache("/root/.pnpm-store", dag.CacheVolume(fmt.Sprintf("pnpm-cache-%s-%s", runtime, version))).
-			// Install tsx
-			WithExec([]string{"npm", "install", "-g", "tsx@4.15.6"}), nil
+			// install tsx from its bundled location in the engine image
+			WithDirectory("/usr/local/lib/node_modules/tsx", t.SDKSourceDir.Directory("/tsx_module")).
```

Comment:
this also works with WithMountedDirectory, and might be a little faster, but i presume mounting like that puts us at risk of different modules overwriting?

Reply:
> but i presume mounting like that puts us at risk of different modules overwriting

by "different modules" do you mean "node modules" or "dagger modules"?

The only risk of WithMountedDirectory I can see is that rather than overlay onto anything existing in `/usr/local/lib/node_modules/tsx`, it will replace that directory entirely.

So if it's a safe assumption that that dir is always empty to start with then I think WithMountedDirectory should be good.

(sorry if that's just begging the question you were asking, the overlapping terminology might be confusing me 😅)

Reply:
I mean dagger modules. Mounts are read-only? Or does each container get its own view of the dir?

Reply:
I think it's fine to use mounted dir, the tsx directory should never change during the execution, it's just a binary binding.
Additional bin or node_modules installation will not impact the `tsx` dir.

Reply:
two ground truths i'm missing, but i trust y'all saying mounts work here and will swap over to that.

@sipsma implicitly, `WithMountedDirectory` gives us a copy-on-write/overlay mount here? fwiw this is what i'd assume, and i feel like i should be able to validate that assumption myself, but beneath [builtinContainer](https://github.com/cwlbraa/dagger/blob/6aed55a063b42a19d78d9c3f564002bfe7e9f8eb/core/schema/host.go?plain=1#L42) there's like a 8 layers of Buildkit abstraction.

@TomChv can users set a different (`-g` global?) version of tsx in their package.json? this is a very hard question to get answers to across 2 runtimes and 4 package managers lol

Reply:
> implicitly, WithMountedDirectory gives us a copy-on-write/overlay mount here? fwiw this is what i'd assume, and i feel like i should be able to validate that assumption myself, but beneath [builtinContainer](https://github.com/cwlbraa/dagger/blob/6aed55a063b42a19d78d9c3f564002bfe7e9f8eb/core/schema/host.go?plain=1#L42) there's like a 8 layers of Buildkit abstraction.

Yes exactly, the directory you provide to withMountedDirectory serves as the base layers and you get a new overlay layer on top, so no possibility of concurrent usage interfering with each other.

Generally that's how everything works except for `withMountedCache`, which is the escape hatch for when you actually do want concurrent users to share the same writable filesystem.